File: exemplos/python/03_catalogo_listar.py
# Primeiros itens do catálogo do inquilino (GET /api/itens, escopo catalogo:ler)
from _apoio import chamar, exigir, exigir_chave

exigir_chave()
status, _cab, corpo = chamar("/api/itens?limite=5")
exigir(status == 200, f"esperava 200 em /api/itens, veio {status}: {corpo}")
itens = corpo.get("itens", corpo if isinstance(corpo, list) else [])
print(f"total no catálogo: {corpo.get('total', len(itens))}; mostrando {len(itens)}")
for item in itens:
    print(f"  {item.get('tipo', '?'):14s} {item.get('titulo', '(sem título)')}")

File: exemplos/js/03_catalogo_listar.mjs
// Primeiros itens do catálogo do inquilino (GET /api/itens, escopo catalogo:ler)
import { chamar, exigir, exigirChave } from "./_apoio.mjs";

exigirChave();
const { status, objeto } = await chamar("/api/itens?limite=5");
exigir(status === 200, `esperava 200 em /api/itens, veio ${status}: ${JSON.stringify(objeto)}`);
const itens = objeto.itens ?? (Array.isArray(objeto) ? objeto : []);
console.log(`total no catálogo: ${objeto.total ?? itens.length}; mostrando ${itens.length}`);
for (const item of itens) console.log(`  ${(item.tipo ?? "?").padEnd(14)} ${item.titulo ?? "(sem título)"}`);
